### ISSUE TITLE
ethminer-0.18.0-rc.0 "Invalid nonce size" at Nicehash

### ISSUE THREAD

**Issue by j2h4u** [ethereum-mining/ethminer#1858]
**Describe the bug**

0.18.* and Nicehash: all shares are rejected.

**To Reproduce**

ethminer.exe --pool stratum://3B****jy.ch4:x@daggerhashimoto.eu.nicehash.com:3353 --cuda --verbosity 3

Same results with "stratum2+tcp".

**Expected behavior**

Previous branches like 0.16 and 0.17 are ok with Nicehash.
0.18-alpha didn't work with NH as well, but error was like "invalid Json message : stoul argument out of range." 

**Screenshots (Optional)**

```
 m 10:28:22 <unknown> 0:01 A0 204.55 Mh - cu0 15.48 A0, cu1 15.91 A0, cu2 15.67 A0, cu3 15.62 A0, cu4 15.92 A0, cu5 15.57 A0, cu6 15.55 A0, cu7 15.64 A0, cu8 15.57 A0, cu9 15.92 A0, cu10 15.92 A0, cu11 15.90 A0, cu12 15.89 A0
 i 10:28:24 <unknown>  << {"params":["000001070910b221","d5c0f3c36d0a543d1ec6363fda4c51ba4ab00e7a5fbfa7d81ebb3ba46fd25dcb","32c8d5c06ee6bb808498711e7b79281468aa91a865fa57550956c24ff40af9a1",true],"id":null,"method":"mining.notify"}
 i 10:28:25 <unknown> Job: 32c8d5c0... daggerhashimoto.eu.nicehash.com [172.65.195.169:3353]
 i 10:28:25 <unknown>  << {"params":["000001070910d457","d5c0f3c36d0a543d1ec6363fda4c51ba4ab00e7a5fbfa7d81ebb3ba46fd25dcb","5096e789b997882e77e738cd4d49e6e6e73df78769b774a5f84a747c0091379c",true],"id":null,"method":"mining.notify"}
 i 10:28:25 <unknown> Job: 5096e789... daggerhashimoto.eu.nicehash.com [172.65.195.169:3353]
cu 10:28:27 cuda-5   Job: 5096e789... Sol: 0x583da52f82d4d129
 i 10:28:27 <unknown>  >> {"id":45,"method":"mining.submit","params":["3B****jy.ch4","000001070910d457","583da52f82d4d129"]}
 **i 10:28:27 <unknown>  << {"id":45,"result":false,"error":[5,"Invalid nonce size.",null]}
 X 10:28:27 <unknown> Reject reason : 5 Invalid nonce size.
 X 10:28:27 <unknown> **Rejected 117 ms. daggerhashimoto.eu.nicehash.com [172.65.195.169:3353]**
```

**Environment (please complete the following information):**

- Operating System: Windows 10
- Hardware: some GTX 1050i

**Comment by keksikSK** [comment]
hello,

i have the same problem

12xRX580 4/8G
Win10 x64

**Comment by j2h4u** [comment]
Linux version doesn't work with Nicehash too. Tested under the HiveOS.

**Comment by madscientist159** [comment]
Seeing this with the latest GIT miner on Talos II / Tesla V100.

```
 X 01:00:32 ethminer Reject reason : 5 Invalid nonce size.
 X 01:00:32 ethminer **Rejected 102 ms. daggerhashimoto.usa.nicehash.com [172.65.194.252:3353]
```

**Comment by slakov** [comment]
same with both 0.18 and 0.19. Linux version.

**Comment by Konstantin35** [comment]
why the difference in nonce
must be 354b100002ded1c7  = 16
issues e32da15bd8869a8db414 = 20

**Comment by AndreaLanfranchi** [comment]
Solved by #1983
